ext/reflection/php_reflection.c

```diff
@@ -5715,6 +5715,106 @@ ZEND_METHOD(reflection_property, hasType)
 }
 /* }}} */
 
+/* {{{ proto public bool ReflectionProperty::hasDefaultValue()
+   Returns whether property has a default value */
+ZEND_METHOD(reflection_property, hasDefaultValue)
+{
+	reflection_object *intern;
+	property_reference *ref;
+	zend_property_info *prop_info;
+	zval *prop, prop_copy;
+	zend_class_entry *ce;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	GET_REFLECTION_OBJECT_PTR(ref);
+
+	prop_info = ref->prop;
+	ce = prop_info->ce;
+
+	prop = NULL;
+	if ((prop_info->flags & ZEND_ACC_STATIC) != 0) {
+		prop = &ce->default_static_members_table[prop_info->offset];
+		ZVAL_DEINDIRECT(prop);
+	} else if ((prop_info->flags & ZEND_ACC_STATIC) == 0) {
+		prop = &ce->default_properties_table[OBJ_PROP_TO_NUM(prop_info->offset)];
+	}
+
+	if (!prop || (ZEND_TYPE_IS_SET(prop_info->type) && Z_ISUNDEF_P(prop))) {
```

Comment:
Checking for a type should not be necessary here.

Reply:
This is still unresolved.

Reply:
@nikic can you clarify why this is unresolved, and then in line 5796 the same code is fine? I copied this from ReflectionClass::getDefaultProperties(), so i am unsure why this is done anyways.

Reply:
Same applies to the code below yes. Probably also to getDefaultProperties(). The !prop check is also redundant.